lib/streams/documentStream.js

```diff
@@ -135,6 +135,15 @@ function processRecord(record, next_uid, stats) {
 
     const pelias_document = new peliasModel.Document( source, layer, model_id );
 
+    const parent = getCaseInsensitiveAsJSON( `parent_json`, record );
+    if (parent) {
```

Comment:
You can make the code a bit more robust and readable with `lodash`, such as this:

```js
if (_.isPlainObject(parent)) {
  _.each(parent, (value, field) => {
    pelias_document.addParent(field, value.name, value.id, value.abbr, value.source)
  })
}
```

It has a couple advantages:
- explicitly check the type of `parent` is an object which prevents a class of errors when the `parent_json` object is incorrectly specified.
- use the `_.each` function to simplify iteration over the properties and make it a bit more succinct and readable

Reply:
In the case where `parent_json` is incorrectly formatted there will be no error returned to the user, this may cause frustration if the import it long-running as it will take time to detect the error and require re-running the import.

Should we detect a situation where the `parent_json` column is provided but contains either malformed or invalid values? In this situation should we panic or emit an error/warning message?

Reply:
**Regarding the validation:**

Right now we are only showing the warnings in the logs and ignore the records which are malformed, this is the similar behaviour as with the other fields. 
Following warning is shown for the malformed records:

`Record import failed on line 26 due to invalid document type, expecting: string got: undefined.`

In addition to that we have implemented the additional validation in the model to check that the parent fields are valid, [the PR is released](https://github.com/pelias/model/pull/150) in `pelias-model@9.4.0`. 
I have upgraded the version of pelias-model in csv-importer as part of this PR, so that if we use the non-existing parent name the record will be ignored with following warning.

`Record import failed on line 36 due to invalid property: land, should be one of: continent,country,dependency,macroregion,region,macrocounty,county,borough,locality,localadmin,neighbourhood,postalcode,ocean,marinearea,empire.`

The way the csv-importer works is, it try to import all the records, give the warnings for the records it fails to import and give the following report at the end of import.

`2022-08-04T12:38:39.408Z - info: [csv-importer] Finished import from /Users/me/1659614476347.csv, goodRecordCount=60371 badRecordCount=326`

The report show that the `warn and not fail` was the intentional implementation. Don't know why it was done like that. 🤔 
This is not specific to `parent_json` field, i think changing this behaviour may affect the other users of csv-importer.

Reply:
makes sense :+1: this sounds sufficient